src/main/java/io/jenkins/plugins/extratoolinstallers/installers/IsAlreadyInGivenDirectory.java

```diff
@@ -0,0 +1,587 @@
+        final String[] splitB = compoundVersionB == null ? new String[0] : compoundVersionB.split("\\.", -1);
+        final int lengthA = splitA.length;
+        final int lengthB = splitB.length;
+        final int highestLength = Math.max(lengthA, lengthB);
+        for (int i = 0; i < highestLength; i++) {
+            final String partA = lengthA > i ? splitA[i] : null;
+            final String partB = lengthB > i ? splitB[i] : null;
+            final int result = compareNullableVersionParts(partA, partB);
+            if (result != 0) {
+                return result;
+            }
+        }
+        return 0;
+    }
+
+    /**
+     * @return 0 if versions are equivalent, +ve if partA is higher than partB, -ve
+     * if partA is lower.
+     */
+    private static int compareNullableVersionParts(@Nullable String partA, @Nullable String partB) {
+        if (partA == null) {
+            if (partB == null) {
+                return 0;
+            } else {
+                return -1;
+            }
+        } else {
+            if (partB == null) {
+                return 1;
+            } else {
+                return compareVersionParts(partA, partB);
+            }
+        }
+    }
+
+    /**
+     * Compares two strings on the assumption that they have a numerical start
+     * followed by lexicographical part.
+     *
+     * @return 0 if versions are equivalent, +ve if a is higher than b, -ve if a is
+     * lower.
+     */
+    private static int compareVersionParts(@Nonnull String a, @Nonnull String b) {
+        final int ai = findIndexOfFirstNonnumericalCharacter(a);
+        final int bi = findIndexOfFirstNonnumericalCharacter(b);
+        final String aNumberString;
+        final String bNumberString;
+        final String aRemainder;
+        final String bRemainder;
+        if (ai >= 0) {
+            aNumberString = a.substring(0, ai);
+            aRemainder = a.substring(ai);
+        } else {
+            aNumberString = a;
+            aRemainder = "";
+        }
+        if (bi >= 0) {
+            bNumberString = b.substring(0, bi);
+            bRemainder = b.substring(bi);
+        } else {
+            bNumberString = b;
+            bRemainder = "";
+        }
+        final long aNumber = aNumberString.isEmpty() ? -1L : Long.parseLong(aNumberString);
+        final long bNumber = bNumberString.isEmpty() ? -1L : Long.parseLong(bNumberString);
+        if (aNumber > bNumber) {
+            return 1;
+        }
+        if (aNumber < bNumber) {
+            return -1;
+        }
+        return aRemainder.compareTo(bRemainder);
+    }
+
+    private static int findIndexOfFirstNonnumericalCharacter(@Nonnull String s) {
+        final int l = s.length();
+        for (int i = 0; i < l; i++) {
+            final char c = s.charAt(i);
+            if (!Character.isDigit(c)) {
+                return i;
+            }
+        }
+        return -l;
+    }
+}
```

Comment:
Please improve code reuse - most of this file is just a copy/paste of IsAlreadyOnPath.
Don't copy/paste the code, define it once and call it from both places.
e.g. the version comparison methods can simply be called without any changes to IsAlreadyOnPath, and many other parts of IsAlreadyOnPath could be made reuseable by a simple refactor.

Reply:
@pjdarton Please review my latest commits. I tried to fit duplicated code to a parent class as much as possible to reduce duplication.